### DISCUSSION TITLE
Another Community Java Autograder Library

### DISCUSSION THREAD

**Discussion by bralax** [gradescope/autograder_samples#49] [General]
Hi! I just wanted to share [another open-source java autograder library](https://github.com/bralax/gradescope_autograder) that was built at Johns Hopkins University, around the same time as JGrade, just for more introductory courses. The library was designed to be able to look a little more closely at the details of students. 

The idea is that in intro classes, professors often ask for very specific and small requirements (like making a method with exact parameter names and types or declaring a specified set of class properties as public/private/protected), so the tests reflect that. 

It also has a lot of little tools/tricks to properly handle the common mistakes/situations that exist in  introductory courses that  could mess with testing results (like when creating multiple scanners connected to standard input or when introducing random number generators to apps).

I just wanted to share it to see what other people think!

**Comment by ibrahima** [comment]
Cool, thanks for sharing! Feel free to submit a PR to add it to https://github.com/gradescope/autograder_samples/blob/master/docs/resources.md
